diagnostic.md

```diff
@@ -5,20 +5,24 @@ Record your responses inside the fenced code blocks below each question.
 1.  Give an example of a visual hierarchy that could be modeled with components.
 
     ```md
-    <!-- your response here -->
+      Any kind of listing app where you have several lists, each of which has
+      several list items rendered within it
     ```
 
 1.  What is the command to generate a new component called '`my-map`'?
 
     ```sh
-    # your response here
+      ember generate component my-map
     ```
 
 1.  What files are edited to produce a component, and what are their
     responsibilities?
 
     ```md
-    <!-- your response here -->
+      The component generator creates a component.js file and a template.hbs
+      file in a folder with the name of the component. The template file is
+      responsible for displaying the content, and the component.js file is
+      responsible for defining the component's behavior, e.g. click handlers
```

Comment:
Yes, looks good.  In addition to markup, Template files are responsible for binding data, and binding and dispatching actions.  In addition to defining actions, Components objects are responsible for declaring HTML attributes.